### PR TITLE
Removed retain cycle from notification center.

### PR DESCRIPTION
**Pull request by gaaclarke**
related PR: https://github.com/flutter/engine/pull/13062

### CODE REVIEW

**Comment by jmagman** [comment]
@gaaclarke and I were talking on chat, looks like the observer needs to be removed.  Will hold off reviewing for now.

**Comment by gaaclarke** [comment]
> @gaaclarke and I were talking on chat, looks like the observer needs to be removed. Will hold off reviewing for now.

Thanks @jmagman I learned 2 things about notification center that I thought I knew.
1. The object returned from addObserver has a retain on the block.
1. The object returned from addObserver still has to be explicitly removed.

Both are not the decisions I would have made *shrug*.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 1]
```diff
@@ -176,15 +179,27 @@ - (void)setViewController:(FlutterViewController*)viewController {
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
 
+  __block FlutterEngine* blockSelf = self;
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm]: 🔮 The retain cycle will be reintroduced when someone finally hits the "modernize to ARC" button without auditing all the blocks where weak memory management is expected. 🔮

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterEngine.mm]: New word, "retro-modernize"?  When we move to ARC it should work as well.  There is a subtle difference between __block and __weak (like how they treat primitives).  For an object pointer it should behave the same.  I'm more sure about that than my understanding of the `addObserverForName:object:queue:usingBlock:` but obviously I've been wrong before, haha.

**Pull request merged**